### PR TITLE
Issue/26 wait after notification

### PR DESCRIPTION
**Pull request by iperdomo**
* Use `overtone/at-at` for delaying the fetch of events
* Remove all references to `core.async`
* Default values if not configured: 15 min delay and 5 threads in the scheduler thread-pool
* Update libraries to most recent ones
* Enhance the `/status` endpoint to show info about scheduled work
* Move the fetch-and-insert functions to `db` namespace

### CODE REVIEW

File: src/akvo_unified_log/config.clj [hunk 1]
```diff
@@ -60,80 +68,25 @@
...
-(defn init-config [repos-dir config-file-name event-notification-handler]
-  (let [config (read-config repos-dir config-file-name)
-        event-notification-chan (async/chan (async/sliding-buffer 1000))
-        event-notification-pub (async/pub event-notification-chan :org-id)]
-    (-> config
-        (assoc :event-notification-chan event-notification-chan
-               :event-notification-pub event-notification-pub
-               :event-notification-handler event-notification-handler)
-        (subscribe-all (keys (:instances config))))))
+(defn init-config [repos-dir config-file-name]
```
Comment (jonase) [inline src/akvo_unified_log/config.clj]: Looking at the function body, this function is identical to `read-config`.

File: src/akvo_unified_log/scheduler.clj [hunk 2]
```diff
@@ -0,0 +1,31 @@
+(ns akvo-unified-log.scheduler
+  (:require [overtone.at-at :as at]))
+
+;; Ideally this should be a Component
```
Comment (jonase) [inline src/akvo_unified_log/scheduler.clj]: Make it a component or remove the comment?

File: src/akvo_unified_log/core.clj [hunk 3]
```diff
@@ -1,91 +1,38 @@
...
+                      :config-file-name config-file-name)]
     (statsd/setup (:statsd-host config)
                   (:statsd-port config)
                   :prefix (:statsd-prefix config))
     (log/merge-config! {:level (:log-level config :info)
                         :output-fn (partial log/default-output-fn
                                             {:stacktrace-fonts {}})})
-    (let [port (Integer. (:port config 3030))
+    (let [_ (scheduler/set-thread-pool (:num-threads config 5))
+          _ (scheduler/set-delay (:fetch-delay config (* 60 5)))
```
Comment (jonase) [inline src/akvo_unified_log/core.clj]: these two could be moved out of the `let` to avoid the `_` binding

File: src/akvo_unified_log/core.clj [hunk 4]
```diff
@@ -1,91 +1,38 @@
...
+            [ring.util.response :refer [not-found]])
+  (:gen-class))
 
 (defn app [config]
   (routes
-   (ANY "/status" _ (endpoints/status config))
-   (ANY "/event-notification" _ (endpoints/event-notification config))
-   (ANY "/reload-config" _ (endpoints/reload-config config))
-   (POST "/events" _ (endpoints/event-push config))))
+   (GET "/favicon.ico" _ (not-found "Not found"))
```
Comment (jonase) [inline src/akvo_unified_log/core.clj]: Do we want this `GET`? Isn't a `404` returned automatically?

Comment (iperdomo) [reply src/akvo_unified_log/core.clj]: Nope, it fails with an exception

File: src/akvo_unified_log/scheduler.clj (hunk 2 above)
Comment (iperdomo) [reply src/akvo_unified_log/scheduler.clj]: Agree. I'll remove the comment

File: src/akvo_unified_log/core.clj (hunk 4 above)
Comment (jonase) [reply src/akvo_unified_log/core.clj]: Oh? What kind of exception?

**Review by jonase — APPROVED** [review]

**Pull request merged**